js/build.js

```diff
@@ -113,101 +96,70 @@ Fliplet.Widget.instance({
+                      if (keyType === 'array') {
+                        isConditionIncluded(user[userKey], condition);
                       } else if (keyType === 'string') {
-                      // check if string can be parsed into JSON array
-                        let ifJSON = ifValidJson(user[conditions[i]['user_key']]);
+                        // check if string can be parsed into JSON array
+                        let parsedType = getParsedType(user[userKey]);
+
+                        if (parsedType === 'array') {
+                          let currentArray = JSON.parse(user[userKey]);
 
-                        if (!ifJSON || ifJSON === 'string') {
-                        // parsed value is not an array nor a string
-                          expression = user[conditions[i]['user_key']]
+                          isConditionIncluded(currentArray, condition);
+                        } else {
+                          expression = user[userKey]
                             .split(',')
                             .map(el => el.trim())
-                            .includes(decodeHTMLEntities(conditions[i]['user_value']));
-                          result = setResult(evaluate(conditions[i], expression));
-                        } else if (ifJSON === 'array') {
-                          let currentArray = JSON.parse(user[conditions[i]['user_key']]);
-
-                          result = ifArrayIncludes(currentArray, conditions[i]);
+                            .includes(decodeHTMLEntities(condition.user_value));
+                          setResult(evaluate(condition, expression));
                         }
-                      } else if (keyType === 'array') {
-                        result = ifArrayIncludes(user[conditions[i]['user_key']], conditions[i]);
+                      } else {
+                        // other type but array or string
+                        expression = '"' + user[userKey] + '".indexOf("' + condition.user_value + '") > -1';
+                        setResult(evaluate(condition, expression));
                       }
                     }
-                  } else if (environment) {
-                    Fliplet.UI.Toast('User doesn\'t contain key: ' + conditions[i]['user_key']);
+                  } else if (isPreview) {
+                    Fliplet.UI.Toast(`User doesn't contain key: ${userKey}`);
                   }
                 }
               }
 
               if (result) {
                 $(helper.el).removeClass('hidden');
               }
-            } else if (environment) {
-              Fliplet.UI.Toast('User is not logged in');
+            } else if (isPreview) {
+              Fliplet.UI.Toast(userNotLoggedMessage);
             }
-          } else if (environment) {
-            Fliplet.UI.Toast('User is not logged in');
+          } else if (isPreview) {
+            Fliplet.UI.Toast(userNotLoggedMessage);
           }
 
           return Promise.resolve(true);
```

Comment:
This is already in a promise chain so Promise.resolve() isn't needed.

```suggestion
          return true;
```